messenger/decoder.go

```diff
@@ -523,6 +524,19 @@ func readBodyState(d *httpDecoder) httpState {
 	return readBodyState
 }
 
+func isGracefulTermSignal(err error) bool {
```

Comment:
Is this name appropriate? AFAIK `ECONNRESET` doesn't mean the connection was *gracefully* closed on the other side, only that it was closed.